docs/contribute/DEV_DOCS.md

```diff
@@ -45,55 +45,48 @@ output of `augur` installed with pip and `./bin/augur` from your local source co
+#### 1. Unit tests
 
-* suitably test a single augur command with an eponymously named Cram file in `tests/functional/` (e.g., `mask.t` for augur mask)
+Unit tests are written using [pytest](https://docs.pytest.org).
+Augur's unit tests are located in the `tests` directory and prefixed with `test_`. There is generally one test file for each code file.
 
-OR
+#### 2. Doctests
 
-* test a complete build with augur commands with an appropriately named Cram file in `tests/builds/` (e.g., `zika.t` for the example Zika build)
+[Doctests](https://docs.python.org/3/library/doctest.html) are a type of test that are written within a module's docstrings.
+They can be helpful for testing a real-world example and determining if a regression is introduced in a particular module. They are run via `pytest`.
 
-##### Functional tests of specific commands
+#### 3. Functional tests
 
-Functional tests of specific commands consist of a single Cram file per test and a corresponding directory of expected inputs and outputs to use for comparison of test results.
+Augur's command line interface is tested by functional tests implemented with the [Cram framework](https://bitheap.org/cram/).
+These tests complement existing unit tests of individual augur Python functions by running augur commands on the shell and confirming that these commands:
 
-The Cram file should test most reasonable combinations of command arguments and flags.
+1. execute without any errors
+2. produce exactly the expected outputs for the given inputs
 
-##### Functional tests of example builds
+These tests can reveal bugs resulting from untested internal functions or untested combinations fo internal functions.
 
-Functional tests of example builds use output from a real Snakemake workflow as expected inputs and outputs.
-These tests should confirm that all steps of a workflow can execute and produce the expected output.
-These tests reflect actual augur usage in workflows and are not intended to comprehensively test interfaces for specific augur commands.
+Over time, we have changed the way we design and organize Augur's Cram tests. You might find older practices in existing tests that haven't been updated yet, but these are the latest guidelines that we've discovered to be helpful.
 
-The Cram file should replicate the example workflow from start to end.
-These tests should use the output of the Snakemake workflow (e.g., files in `zika/results/` for the Zika build test) as the expected inputs and outputs.
+1. Keep cram files modular. This makes it easier to see which command is failing.
+2. Create files in the initial working directory, as it is a temporary working directory unique to the test. Note that the name of the `$TMP` directory is misleading - although it is temporary, it is shared across all tests so you'll have to explicitly remove files at the end of each test to avoid affecting other tests. The initial directory of each test is a unique directory within `$TMP`.
```

Comment:
Is it worth noting how one accesses the "initial directory" in a portable way? Is it just that paths should have a `./` prefix?

Reply:
Done: f785a7e9eade7d3a91455722a3b8ebcbdace29f8

> Create files in the initial working directory (e.g. `./file.txt` or simply `file.txt`)